tests/config/test_omegaconf_config.py

```diff
@@ -401,6 +401,7 @@ def test_empty_catalog_file(self, tmp_path):
         )["catalog"]
         assert catalog == {}
 
+    @pytest.mark.xfail(reason="Logic failing")
```

Comment:
Fixed the test to check for the intended behaviour. However, it fails, and manual testing shows that this is due to the implementation of the OmegaConfigLoader. I would suggest correcting this should be split out into a separate issue to be addressed.

Reply:
I don't agree with this being addressed in a separate issue, adding the `3.12` support is what's making this fail and thus it should be fixed in this PR.

Reply:
It's not the move to include 3.12 that's causing this test to fail, but rather it was never truly passing, the previous method call used for the assert was mistyped, leading to a trivial pass. To resolve this requires digging further into the implementation of the OmegaConfigLoader, as it is not behaving as expected, and perhaps never was. See [xref](https://github.com/kedro-org/kedro/pull/3587#discussion_r1476292680).